### PR TITLE
Imp tests

### PR DESCRIPTION
**Pull request by laurayuwen**
This PR updates all tests (config and .out) in imp and imp++ (except for search, I exclude those tests and did not touch those .out)

### CODE REVIEW

**Comment by grosu** [comment]
@laurayuwen , I hope you used the ktest tool to automatically update all the .out files; it has an option exactly for this purpose, because we updated these many times in the past and we are still going to update them many times from here on.  We don't want to do this manually.

File: k-distribution/tutorial/1_k/2_imp/lesson_1/tests/collatz.imp.out [hunk 1]
```diff
@@ -1,5 +1 @@
-<k>
-    int m, n, q, r, s ; m = 10 ; while ( ! m <= 2 ) { n = m ; m = m + -1 ; 
-      while ( ! n <= 1 ) { s = s + 1 ; q = n / 2 ; r = q + q + 1 ; if ( r <= 
-      n ) { n = n + n + n + 1 ; } else { n = q ; } } }
-</k>
+<k> int m , ( n , ( q , ( r , ( s , .Ids ) ) ) ) ; ( m = 10 ; while ( ! ( m <= 2 ) ) { n = m ; m = m + -1 ; while ( ! ( n <= 1 ) ) { s = s + 1 ; q = n / 2 ; r = q + q + 1 ; if ( r <= n ) { n = n + n + n + 1 ; } else { n = q ; } } } ) </k>
```
Comment (grosu) [inline k-distribution/tutorial/1_k/2_imp/lesson_1/tests/collatz.imp.out:1]: @andreistefanescu , are we going to also get rid of `int m , ( n , ( q , ( r , ( s , .Ids ) ) ) )` once we merge your A/AC RP?  I mean, are the user-defined lists assumed associative in the semantics?  If no, can we make them or it is too risky?

**Comment by laurayuwen** [comment]
@grosu do you mean --update-out? I used that.

**Comment by grosu** [comment]
Yes, that's what I meant.  Great!  I'm assuming it worked well, but if it didn't, please feel free to report issues with it or suggest improvements.  I think you may be the only one who used it besides @daejunpark  and myself.

**Comment by laurayuwen** [comment]
@daejunpark taught me this :) It works great for me. But I still go lesson by lesson to check the config and so on, once I am happy with a lesson, I use this option to update.

File: k-distribution/tutorial/1_k/2_imp/lesson_1/tests/collatz.imp.out (hunk 1 above)
Comment (andreistefanescu) [inline k-distribution/tutorial/1_k/2_imp/lesson_1/tests/collatz.imp.out:1]: Currently, the used-defined lists are not treated as associative. We can make them associative, we only need to extend the existing A/AC pr to work when the assoc operator is overloaded (List{Exp} and List{Int}).

Comment (dwightguth) [inline k-distribution/tutorial/1_k/2_imp/lesson_1/tests/collatz.imp.out:1]: Just an FYI, I would strongly prefer user defined lists to only be
associative if they are explicitly given the assoc attribute.

**Comment by laurayuwen** [comment]
Jenkins fails because of copyright check: the files are now 2016 but strangely expect 2015. 
I am closing this PR and create a new PR towards master directly since imp definitions have already been merged into master.

**Pull request closed without merging**